### PR TITLE
Add call to HTML::checkRelMeLink in Worker\CheckRelMeProfileLink

### LINKED ISSUE #12507: Homepage verification inconsistent?
Some links don't seem to pass verification procedure and checkmark doesn't appear.

Test profile:
https://friends.deko.cloud/profile/bot

Link examples:
https://social.deko.cloud/test.html - passes verification
https://lor.sh/@shuro - doesn't pass verification

When the second link is set in the profile the following is registered in the log:
```
2022-12-22T16:42:31Z worker [INFO]: Process start. {"priority":30,"id":61906637,"worker_id":"0677fea","worker_cmd":"CheckRelMeProfileLink"} - {"file":"Worker.php","line":554,"function":"execFunction","uid":"cf1176","process_id":1321120}
2022-12-22T16:42:31Z worker [NOTICE]: Verifying the homepage {"uid":2,"worker_id":"0677fea","worker_cmd":"CheckRelMeProfileLink"} - {"file":"CheckRelMeProfileLink.php","line":57,"function":"execute","uid":"cf1176","process_id":1321120}
2022-12-22T16:42:31Z worker [NOTICE]: Homepage URL could not be verified {"uid":2,"owner homepage":"https://lor.sh/@shuro","worker_id":"0677fea","worker_cmd":"CheckRelMeProfileLink"} - {"file":"CheckRelMeProfileLink.php","line":88,"function":"execute","uid":"cf1176","process_id":1321120}
2022-12-22T16:42:31Z worker [INFO]: ID 61906637: CheckRelMeProfileLink {"action":"profiling","database_read":0,"database_write":0,"cache_read":0,"cache_write":0,"network_io":0,"file_io":0,"other_io":0.17,"total":0.17} - {"file":"Worker.php","line":627,"function":"execFunction","uid":"cf1176","process_id":1321120}
2022-12-22T16:42:31Z worker [INFO]: ID 61906637: CheckRelMeProfileLink:  {"action":"profiling"} - {"file":"Worker.php","line":627,"function":"execFunction","uid":"cf1176","process_id":1321120}
```

However if I curl links they seem to be okay except minor syntax differences:
```
curl https://social.deko.cloud/test.html | grep bot
```
```
<a href="https://friends.deko.cloud/profile/bot" rel="me">test</a>
```
```
100<link href='https://friends.deko.cloud/profile/bot' rel='me' type='text/html'>
```

Another example with another profile and link:
```
curl  https://ro.gravatar.com/cata195  | grep libra
```
```
<a class="profile-link" href="https://libranet.de/profile/petrescatraian" target="_blank" rel="me nofollow noopener">libranet.de/profile/petrescatraian</a>
```

### PR DESCRIPTION
**Pull request by MrPetovan**
Ok, maybe not that Junior of a job. 😅

Fixes #12507

### CODE REVIEW

File: src/Content/Text/HTML.php [hunk 1]
```diff
@@ -1007,4 +1008,51 @@ public static function purify(string $text, array $allowedIframeDomains = []): s
...
+	/**
+	 * Checks if the provided URL is present in the DOM document in an element with the rel="me" attribute
+	 *
+	 * XHTML Friends Network http://gmpg.org/xfn/
+	 *
+	 * @param DOMDocument  $doc
+	 * @param UriInterface $meUrl
+	 * @return bool
+	 */
+	public static function checkRelMeLink(DOMDocument $doc, UriInterface $meUrl): bool
```
Comment (nupplaphil) [inline src/Content/Text/HTML.php:1046]: Seems perfect for unit tests ;-)

File: src/Content/Text/HTML.php [hunk 2]
```diff
@@ -1007,4 +1008,51 @@ public static function purify(string $text, array $allowedIframeDomains = []): s
...
 	}
+
+	/**
+	 * XPath arbitrary string quoting
+	 *
+	 * @see https://stackoverflow.com/a/45228168
+	 * @param string $value
+	 * @return string
+	 */
+	public static function xpathQuote(string $value): string
```
Comment (nupplaphil) [inline src/Content/Text/HTML.php:1019]: Seems perfect for unit tests

**Review by nupplaphil — CHANGES_REQUESTED** [review]
Teeeeests :-)

File: src/Content/Text/HTML.php (hunk 1 above)
Comment (MrPetovan) [reply src/Content/Text/HTML.php:1046]: Please scroll down the diffs 🎅

File: src/Content/Text/HTML.php (hunk 2 above)
Comment (MrPetovan) [reply src/Content/Text/HTML.php:1019]: Please scroll down the diffs 🎅

**Pull request merged**